detection_rules/navigator.py

```diff
@@ -162,11 +162,19 @@ def links_dict(label: str, url: any) -> dict:
         return links
 
     def rule_links_dict(self, rule: TOMLRule) -> dict:
+        """Create a links dictionary for a rule."""
         base_url = 'https://github.com/elastic/detection-rules/blob/main/rules/'
-        try:
-            base_path = str(rule.path.resolve().relative_to(DEFAULT_PREBUILT_RULES_DIR))
-        except ValueError:
-            base_path = str(rule.path.resolve().relative_to(DEFAULT_PREBUILT_BBR_DIR))
+        base_path = None
+        for rules_dir in DEFAULT_PREBUILT_RULES_DIRS + DEFAULT_PREBUILT_BBR_DIRS:
```

Comment:
Same logic in packaging.py. Maybe we can consolidate to a rule method?

Reply:
Agreed :+1:

Reply:
Added a function in TOMLRule called `get_rules_dir_path` I did not make it a property due to the `path` field in TOMLRule being optional, but could make it a property if prefered.